metronome/hotstuff/service/test/src/io/iohk/metronome/hotstuff/service/sync/ViewSynchronizerProps.scala

```diff
@@ -0,0 +1,374 @@
+        syncEventsRef.update(_ :+ event)
+      }
+
+    implicit val syncTracers: SyncTracers[Task, TestAgreement] =
+      SyncTracers(syncEventTracer)
+
+    def getStatus(
+        publicKey: TestAgreement.PKey
+    ): Task[Option[Status[TestAgreement]]] =
+      for {
+        round <- responseCounterRef.modify { responseCounter =>
+          val count = responseCounter(publicKey)
+          responseCounter.updated(publicKey, count + 1) -> count
+        }
+        result =
+          if (round >= responses.size) None
+          else
+            responses(round)(publicKey) match {
+              case TestResponse.Timeout               => None
+              case TestResponse.InvalidStatus(status) => Some(status)
+              case TestResponse.ValidStatus(status)   => Some(status)
+            }
+      } yield result
+  }
+
+  object TestFixture {
+    implicit val leaderSelection = LeaderSelection.RoundRobin
+
+    implicit val arb: Arbitrary[TestFixture] = Arbitrary {
+      for {
+        state <- genInitialState
+        federation = Federation(state.federation, state.f).getOrElse(
+          sys.error("Invalid federation.")
+        )
+        byzantineCount <- Gen.choose(0, state.f)
+        byzantines = federation.publicKeys.take(byzantineCount).toSet
+        rounds <- Gen.posNum[Int]
+        genesisQC = state.newViewsHighQC
+        responses <- genResponses(rounds, federation, byzantines, genesisQC)
+      } yield TestFixture(
+        rounds,
+        federation,
+        responses
+      )
+    }
+  }
+
+  sealed trait TestResponse
+  object TestResponse {
+    case object Timeout                                     extends TestResponse
+    case class ValidStatus(status: Status[TestAgreement])   extends TestResponse
+    case class InvalidStatus(status: Status[TestAgreement]) extends TestResponse
+  }
+
+  /** Generate a series of hypothetical responses projected from an idealized consensus process. */
+  def genResponses(
+      rounds: Int,
+      federation: Federation[TestAgreement.PKey],
+      byzantines: Set[TestAgreement.PKey],
+      genesisQC: QuorumCertificate[TestAgreement]
+  ): Gen[Responses] = {
+
+    def genPrepareQC(qc: QuorumCertificate[TestAgreement]) =
+      for {
+        quorumKeys <- Gen
+          .pick(federation.quorumSize, federation.publicKeys)
+          .map(_.toVector)
+        blockHash <- genHash
+        viewNumber = qc.viewNumber.next
+        partialSigs = quorumKeys.map { publicKey =>
+          val signingKey = mockSigningKey(publicKey)
+          Signing[TestAgreement].sign(
+            signingKey,
+            Phase.Prepare,
+            viewNumber,
+            blockHash
+          )
+        }
+        groupSig = mockSigning.combine(partialSigs)
+      } yield QuorumCertificate[TestAgreement](
+        Phase.Prepare,
+        viewNumber,
+        blockHash,
+        groupSig
+      )
+
+    def genCommitQC(qc: QuorumCertificate[TestAgreement]) =
```

Comment:
`genCommitQC` and `genPrepareQC` are very similar. Could we DRY them?

Reply:
Certainly! No idea why I left it like this.

Reply:
Done, thanks!